engine/src/ast/function_expr.rs

```diff
@@ -92,21 +106,25 @@ impl<'s> FunctionCallExpr<'s> {
     }
 
     pub fn execute(&self, ctx: &'s ExecutionContext<'s>) -> LhsValue<'_> {
-        self.function.implementation.execute(
-            self.args.iter().map(|arg| arg.execute(ctx)).chain(
-                self.function.opt_params[self.args.len() - self.function.params.len()..]
-                    .iter()
-                    .map(|opt_arg| opt_arg.default_value.as_ref()),
+        self.function.execute(
+            &mut self.args.iter().map(|arg| arg.execute(ctx)).chain(
+                (self.args.len()
+                    ..self
```

Comment:
I find this difficult to read, could we split up the arg count calculation into a separate binding at the start of the function?

Reply:
pub fn execute(&self, ctx: &'s ExecutionContext<'s>) -> LhsValue<'_> {
        let max_args = self
            .function.max_arg_count()
            .unwrap_or_else(|| self.args.len());
        self.function.execute(
            &mut self.args.iter().map(|arg| arg.execute(ctx)).chain(
                (self.args.len()..max_args)
                    .map(|index| self.function.default_value(index).unwrap()),
            ),
        )
    }